### PR TITLE
Python: switch to shared implementation of IncompleteHostnameRegExp.ql

### PR DESCRIPTION
**Pull request by aibaars**

### CODE REVIEW

File: python/ql/lib/semmle/python/RegexTreeView.qll [hunk 1]
```diff
@@ -1009,3 +1028,24 @@ class RegExpBackRef extends RegExpTerm, TRegExpBackRef {
 
 /** Gets the parse tree resulting from parsing `re`, if such has been constructed. */
 RegExpTerm getParsedRegExp(StrConst re) { result.getRegex() = re and result.isRootTerm() }
+
+/**
+ * A node whose value may flow to a position where it is interpreted
```
Comment (aibaars) [inline python/ql/lib/semmle/python/RegexTreeView.qll:1033]: @tausbn You asked for this class to me moved to another location. Could you tell me the preferred place?

Comment (tausbn) [reply python/ql/lib/semmle/python/RegexTreeView.qll:1033]: I would just create a file in `lib/semmle/python/dataflow/new` called `Regex.qll`, and import it appropriately. Would that be okay?

**Review by yoff — COMMENTED** [review]
This looks good to me now, although there are a bunch of QL warnings.
Really nice to see this sharing!

**Comment by aibaars** [comment]
@yoff @tausbn I addressed your comments and fixed the QL warnings.

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**